### PR TITLE
Declare type for Array.fromAsync

### PR DESCRIPTION
**Pull request by turbocrime**
this is already in chrome, but not known to typescript.

it fully replaces streamToPromise.

### CODE REVIEW

File: packages/polyfills/streamToPromise.test.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { beforeEach, describe, expect, test } from 'vitest';
+
+import Array from './Array.fromAsync';
+
+// eslint-disable-next-line @typescript-eslint/unbound-method
+const streamToPromise = Array.fromAsync;
+
+describe('streamToPromise()', () => {
```
Comment (jessepinho) [inline packages/polyfills/streamToPromise.test.ts:8]: Hm, I feel like these tests are no longer needed — even the ones below under `old streamToPromise behavior`. You're essentially testing something that's part of the Chrome JavaScript API, which should already be well-tested and guaranteed to work. No need to retest it ourselves.

If anything, if there are places where we're using `Array.fromAsync` that _does_ depend on old `streamToPromise` behavior, we should write tests to make sure it's accounted for the fact that `Array.fromAsync` works differently.

File: packages/polyfills/Array.fromAsync.test.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { expect, describe, it } from 'vitest';
+
+import Array from './Array.fromAsync';
+import './ReadableStream[Symbol.asyncIterator]';
+
+describe('Array.fromAsync', () => {
```
Comment (jessepinho) [inline packages/polyfills/Array.fromAsync.test.ts:6]: Tests not needed

**Review by jessepinho — CHANGES_REQUESTED** [review]
Added a couple comments — it seems that the tests you have here aren't really needed, since they're retesting a core API. And can we just redefine `Array` globally, rather than re-exporting it and thus having to import a specific "implementation" of it?

**Comment by turbocrime** [comment]
> it seems that the tests you have here aren't really needed, since they're retesting a core API.

it was useful to demonstrate

1) the actual behavior of that api, since i always get questions every time i do something like this
2) the behavior of the api matches our use-case, by applying the old streamToPromise tests to the browser api

File: packages/polyfills/Array.fromAsync.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+type ArrayWithAsync = typeof Array & {
+  fromAsync<T, U = T>(
+    arrayLike: ArrayLike<T> | AsyncIterable<T>,
+    mapfn?: (v: T, k: number) => U,
+    thisArg?: any,
+  ): Promise<U[]>;
+};
+
+export default Array as ArrayWithAsync;
```
[the remark this answers is not in the document]
Comment (turbocrime) [reply packages/polyfills/Array.fromAsync.ts:12]: typescript exposes `Array` as a `var`, which does not support declaration merging

File: packages/polyfills/streamToPromise.test.ts (hunk 1 above)
Comment (turbocrime) [reply packages/polyfills/streamToPromise.test.ts:8]: maintaining the old tests was useful to demonstrate the api satisfies the behavior we needed

**Comment by turbocrime** [comment]
not making changes

**Review by jessepinho — APPROVED** [review]

**Pull request merged**